version: '3.3'

services:

  rabbitmq:
    image: "sammlerio/rabbitmq"
    container_name: rabbitmq
    # Seems to throw an error on Mac-OS
#    volumes:
#      - ./.data/rabbitmq/data:/var/lib/rabbitmq
#      - ./.data/rabbitmq/log:/var/log/rabbitmq/sammler
    ports:
      - "15672:15672"
      - "5672:5672"
    environment:
      - RABBITMQ_NODENAME=sammler
      - DOCKER_LABELS=true
      - LOGSPOUT=ignore
    labels:
      - io.sammler.env=DEV
      - io.sammler.service=rabbitmq
    depends_on:
      - logstash
    restart: always

  log-service:
    image: "sammlerio/log-service"
    container_name: log-service
    environment:
      - NOVE_ENV=production
      - SERVICE_NAME=log-service
      - MONG

  strategy-heartbeat:
    image: "sammlerio/strategy-heartbeat"
    container_name: strategy-heartbeat
    environment:
      - NODE_ENV=production
      - SERVICE_NAME=strategy-heartbeat
      - RABBITMQ_URI=amqp://guest:guest@s5r-rabbitmq:5672
      - LOG_SERVICE_URI=http://log-service:3004
    volumes:
      - ./src:/opt/strategy-heartbeat
